`SideNav` - Update docs & showcase to add `hds-side-nav-hide-when-minimized` class examples (HDS-4544) (#2719)
hashicorp/design-system: website/docs/components/side-nav/partials/code/component-api.md

  </C.Property>
  <C.Property @name="isResponsive" @type="boolean" @default="true">
    Controls whether the Side Nav is responsive to viewport changes. It can be programmatically turned off by passing `false`.
    <br><br>
    <em>Notice: even if the `@isResponsive` parameter is set to false, some JavaScript is still executed in the background, and event listeners are attached to some DOM elements (even if this functionality is not used).</em>
  </C.Property>
  <C.Property @name="isCollapsible" @type="boolean" @default="false">
    Controls whether the Side Nav is collapsible on large viewports. When this argument and `isResponsive` are set to `true`, a toggle button will permanently be rendered to collapse and expand the Side Nav.
    <br><br>

>     <em>Notice: if `@isResponsive` is set to false, this argument has no effect. If you are not injecting content via portals, you can apply the special `hds-side-nav-hide-when-minimized` class name to the top-level elements of your content to have it automatically fade in/out when the Side Nav changes its "minimization" state.</em>

Reviewer: what do you think of this change?
```suggestion
    <em>Notice: if `@isResponsive` is set to false, this argument has no effect. If you are not injecting content via portals, and you want your content to automatically fade in/out when the Side Nav changes its "minimization" state, you will have to have apply the special `hds-side-nav-hide-when-minimized` class name to the top-level elements of the content you're adding.</em>
```
